Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/AppConfigProvider.kt

```diff
@@ -1,149 +1,45 @@
-            downloadAppConfig()
-        } catch (e: Exception) {
-            Timber.w(e, "Failed to download latest AppConfig.")
-            if (configStorage.isAppConfigAvailable()) {
-                null
-            } else {
-                Timber.e("No fallback available, rethrowing!")
-                throw e
-            }
-        }
-
-        val newConfigParsed = try {
-            tryParseConfig(newConfigRaw)
-        } catch (e: Exception) {
-            Timber.w(e, "Failed to parse latest AppConfig.")
-            null
-        }
-
-        return newConfigParsed?.also {
-            Timber.d("Saving new valid config.")
-            Timber.v("New Config.supportedCountries: %s", it.supportedCountriesList)
-            configStorage.setAppConfigRaw(newConfigRaw)
-        }
-    }
-
-    private suspend fun getFallback(): ApplicationConfiguration {
-        val lastValidConfig = tryParseConfig(configStorage.getAppConfigRaw())
-        return if (lastValidConfig != null) {
-            Timber.d("Using fallback AppConfig.")
-            lastValidConfig
-        } else {
-            Timber.e("No valid fallback AppConfig available.")
-            throw ApplicationConfigurationInvalidException()
-        }
-    }
-
-    suspend fun getAppConfig(): ApplicationConfiguration = mutex.withLock {
-        withContext(Dispatchers.IO) {
-
-            val newAppConfig = getNewAppConfig()
-
-            return@withContext if (newAppConfig != null) {
-                newAppConfig
-            } else {
-                Timber.w("No new config available, using last valid.")
-                getFallback()
-            }
-        }.performSanityChecks()
-    }
+    val currentConfig: Flow<ConfigData?> = currentConfigInternal
 
     suspend fun clear() = mutex.withLock {
-        withContext(Dispatchers.IO) {
-            configStorage.setAppConfigRaw(null)
-
-            // We are using Dispatchers IO to make it appropriate
-            @Suppress("BlockingMethodInNonBlockingContext")
-            cache.evictAll()
-        }
+        Timber.tag(TAG).v("clear()")
+        source.clear()
+        currentConfigInternal.value = null
     }
 
-    private fun ApplicationConfiguration.performSanityChecks(): ApplicationConfiguration {
-        var sanityChecked = this
-
-        if (sanityChecked.supportedCountriesList == null) {
-            sanityChecked = sanityChecked.toNewConfig {
-                clearSupportedCountries()
-                addAllSupportedCountries(emptyList<String>())
-            }
-        }
-
-        val countryCheck = sanityChecked.supportedCountriesList
-        if (countryCheck.size == 1 && !VALID_CC.matches(countryCheck.single())) {
-            Timber.w("Invalid country data, clearing. (%s)", this.supportedCountriesList)
-            sanityChecked = sanityChecked.toNewConfig {
-                clearSupportedCountries()
+    suspend fun getAppConfig(): ConfigData = mutex.withLock {
+        Timber.tag(TAG).v("getAppConfig()")
```

Comment:
Could be removed IMHO

Reply:
I like a bit more verbose logging :blush: